backend/editor/api.py

```diff
@@ -41,16 +44,26 @@
 @app.on_event("startup")
 async def startup():
     """
-    Initialize database
+    Initialize Neo4J database
     """
-    initialize_db()
+    global driver, session
+    uri = settings.uri
+    driver = GraphDatabase.driver(uri)
 
 @app.on_event("shutdown")
 async def shutdown():
     """
-    Shutdown database
+    Close driver of Neo4J database
     """
-    shutdown_db()
+    driver.close()
+
+@app.middleware("http")
+async def transaction_creation(request: Request, call_next):
+    with driver.session() as session:
+        global txn

```

Comment:
give at better name :-) (I know you are still experimenting) like current_transaction